openedge-plugin/src/main/java/org/sonar/plugins/openedge/foundation/IIdProvider.java

```diff
@@ -19,15 +19,16 @@
  */
 package org.sonar.plugins.openedge.foundation;
 
-import org.sonar.plugins.openedge.api.LicenceRegistrar;
+public interface IIdProvider {
 
-public class OpenEdgeLicenceRegistrar implements LicenceRegistrar {
   /**
-   * Register the classes that will be used to instantiate checks during analysis.
+   * @return Permanent ID
    */
-  @Override
-  public void register(Licence registrarContext) {
-    registrarContext.registerLicence("", "", "", "", LicenceRegistrar.LicenceType.EVALUATION, new byte[] {},
-        1451602800000L);
-  }
+  String getPermanentID();
+
+  /**
+   * TODO Remove as soon as SQ 5.x support is dropped, and use context.getRuntime()
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1135)